tests/test_structured.cpp

```diff
@@ -0,0 +1,239 @@
+#ifndef SPDLOG_NO_STRUCTURED_SPDLOG
+
+#include "includes.h"
+#include "test_sink.h"
+
+#include <limits>
+
+#include "spdlog/details/log_msg_buffer.h"
+
+using spdlog::F;
+
+
+std::string log_info(std::initializer_list<spdlog::Field> fields, spdlog::string_view_t what)
+{
+    std::ostringstream oss;
+    auto oss_sink = std::make_shared<spdlog::sinks::ostream_sink_mt>(oss);
+
+    spdlog::logger oss_logger("oss", oss_sink);
+    oss_logger.set_pattern("%v%V");
+    oss_logger.log({}, spdlog::level::info, fields, what);
+
+    return oss.str().substr(0, oss.str().length() - strlen(spdlog::details::os::default_eol));
+}
+
+TEST_CASE("fields", "[structured]")
+{
+    // Can we construct fields of all types with lvalues?
+    F("var", 1);
+    F("var", "val");
+
+    // Can we construct fields with rvalues?
+    std::string str1("str");
+    spdlog::F(str1, str1);
+
+    // Can we construct fields with rvalues?
+    const char * cstr = "str";
+    auto cstr_f = spdlog::F(cstr, cstr);
+    REQUIRE(cstr_f.value_type == spdlog::FieldValueType::INT);
```

Comment:
I'll add additional comments.  There was a lot of fighting with the compiler.  It's super-useful to have literals go to the `string_view` constructor, with a compile-time size calculation.  That's implemented with the `const char (&val)[N]` constructor.  

If I include a `const char *` ctor, string literals will bind closer to that, and we'll have to calculate the string size at runtime.  Since I expect the vast majority of strings to be passed in as either `std::string` or string literals, I thought this was a good starting spot.

I'd like to eventually ensure that both compile-time sizes for string literals, run-time O(1) sizes for `std::strings`, and dynamic sizes for `const char *` are all supported.